cmd/kueuectl/app/list/list_workload.go

```diff
@@ -0,0 +1,311 @@
+	"sigs.k8s.io/kueue/cmd/kueuectl/app/util"
+	"sigs.k8s.io/kueue/pkg/workload"
+)
+
+const (
+	wlLong    = `Lists Workloads that match the provided criteria.`
+	wlExample = `  # List Workload 
+  kueuectl list workload`
+)
+
+type WorkloadOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	OnlyPending        bool
+	OnlyAdmitted       bool
+	AllNamespaces      bool
+	Namespace          string
+	FieldSelector      string
+	LabelSelector      string
+	ClusterQueueFilter string
+	LocalQueueFilter   string
+
+	ClientSet clientset.Interface
+
+	ToPrinter func(r *listWorkloadResources) (printers.ResourcePrinterFunc, error)
+
+	genericiooptions.IOStreams
+}
+
+func NewWorkloadOptions(streams genericiooptions.IOStreams) *WorkloadOptions {
+	return &WorkloadOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewWorkloadCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewWorkloadOptions(streams)
+
+	cmd := &cobra.Command{
+		Use: "workload [-–clusterqueue CLUSTER_QUEUE_NAME] [-–localqueue LOCAL_QUEUE_NAME] [--only-pending] [--only-admitted] [--selector key1=value1] [--field-selector key1=value1] [--all-namespaces]",
+		// To do not add "[flags]" suffix on the end of usage line
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"wl"},
+		Short:                 "List Workload",
+		Long:                  wlLong,
+		Example:               wlExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter, cmd, args))
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addAllNamespacesFlagVar(cmd, &o.AllNamespaces)
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addClusterQueueFilterFlagVar(cmd, &o.ClusterQueueFilter)
+	addLocalQueueFilterFlagVar(cmd, &o.LocalQueueFilter)
+
+	cmd.Flags().BoolVar(&o.OnlyPending, "only-pending", false,
+		"Filter only pending workload.")
+	cmd.Flags().BoolVar(&o.OnlyAdmitted, "only-admitted", false,
+		"Filter only admitted workload.")
+
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *WorkloadOptions) Complete(clientGetter util.ClientGetter, cmd *cobra.Command, args []string) error {
+	var err error
+
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.ClientSet, err = clientGetter.KueueClientSet()
+	if err != nil {
+		return err
+	}
+
+	o.ToPrinter = func(r *listWorkloadResources) (printers.ResourcePrinterFunc, error) {
+		if !o.PrintFlags.OutputFlagSpecified() {
+			printer := newWorkloadTablePrinter().WithResources(r)
+			if o.AllNamespaces {
+				printer.WithNamespace()
+			}
+			return printer.PrintObj, nil
+		} else {
+			printer, err := o.PrintFlags.ToPrinter()
+			if err != nil {
+				return nil, err
+			}
+			return printer.PrintObj, nil
+		}
```

Comment:
```suggestion
		}
		printer, err := o.PrintFlags.ToPrinter()
		if err != nil {
			return nil, err
		}
		return printer.PrintObj, nil
```

This "else" could be eliminated.